qiskit_ibm_runtime/runtime_job.py

```diff
@@ -220,7 +220,7 @@ def result(  # pylint: disable=arguments-differ
         if self._results is None or (_decoder != self._final_result_decoder):
             self.wait_for_final_state(timeout=timeout)
             if self._status == JobStatus.ERROR:
-                error_message = self.error_message()
+                error_message = self._reason if self._reason else self._error_message
                 if self._reason == "RAN TOO LONG":
```

Comment:
Do we want to do something similar when `job.error_message()` is called? We could add the same logic to `_set_error_message()`

Reply:
I am not sure. I don't think we should modify `self._error_message`. Do you think `job.error_message` should return the `_reason` rather than the entire `_error_message`?
In any case, the only place I saw this method called is in the tests.

Reply:
Ah ok I see - yeah `result()` can return the reason and `error_message` can still return the full error message